modules/05-firmware.yml

```
@@ -1,16 +1,35 @@
-name: input-and-locale
-type: apt
-source:
-  packages:
-  - firmware-linux
-  - firmware-linux-nonfree
-  - firmware-linux-free
-  - firmware-iwlwifi
-  - firmware-realtek
-  - firmware-atheros
-  - intel-microcode
-  - amd64-microcode
-  - b43-fwcutter
-  - firmware-b43-installer
-  - firmware-brcm80211
-  - firmware-sof-signed
+name: divert-originals
+type: shell
+commands:
+  - for file in $(find /usr/lib/firmware/ -type f); do dpkg-divert --add --rename --divert "/usr/lib/firmware/${file}-old" "/usr/lib/firmware/${file}"; done
```

```suggestion
  - for file in $(find /usr/lib/firmware/ -type f); do dpkg-divert --add --rename --divert "/usr/lib/firmware/${file}-old" "/usr/lib/firmware/${file}"; done
  - find /usr/lib/firmware -type f -exec dpkg-divert --add --rename --divert '{}-old' '{}' \;
```

using a single `find` command here may be better since it doesn't rely on a bash for loop

---

Nice one